gcp.go

```diff
@@ -126,17 +133,39 @@ type logger interface {
 }
 
 type gcpLogger struct {
-	lg          logger
-	traceID     string
-	mu          sync.Mutex
-	maxSeverity logging.Severity
-	logCount    int
+	root          *gcpLogger
+	logger        logger
+	traceID       string
+	rsvdKeys      []string
+	attributes    map[string]any // attributes for child (trace) logs
+	mu            sync.Mutex
+	maxSeverity   logging.Severity
+	logCount      int
+	reqAttributes map[string]any // attributes for the parent request log
 }
 
 func newGCPLogger(lg logger, traceID string) *gcpLogger {
+	l := &gcpLogger{
+		logger:        lg,
+		traceID:       traceID,
+		rsvdKeys:      []string{gcpMessageKey},
+		reqAttributes: make(map[string]any),
+		attributes:    make(map[string]any),
+	}
+	l.root = l // root is self
+
+	return l
+}
+
+// newChild returns a new child gcpLogger
+func (l *gcpLogger) newChild() *gcpLogger {
 	return &gcpLogger{
-		lg:      lg,
-		traceID: traceID,
+		root:          l.root,
+		logger:        l.logger,
+		traceID:       l.traceID,
+		rsvdKeys:      l.rsvdKeys,
+		reqAttributes: make(map[string]any),
```

Comment:
The child logger being returned here should NEVER use `reqAttributes`. If I am correct in that statement, I wonder if we should leave this nil here. This would be help catch any bugs, document clear intent, and provide a minor performance improvement.